piwik-sdk/src/main/java/org/piwik/sdk/CustomVariables.java

```diff
@@ -73,21 +72,25 @@ public JSONArray put(int index, String name, String value) {
      * @param values packed key/value pair
      * @return super.put result or null if key is null or value length is not equals 2
      */
-    @Override
     public JSONArray put(String index, JSONArray values) {
         if (values.length() == 2 && index != null) {
-            return super.put(index, values);
+            synchronized (mConcurrentLock) {
```

Comment:
Just a different style of coding, do you like it?
(Assuming the change to ConcurrentHashMap)

```
public JSONArray put(String index, JSONArray values) {
    if (values.length() != 2 || index == null) {
        Logy.d(LOGGER_TAG, "values.length() should be equal 2");
    	return null;
    }
    return mVars.put(index, values);
}
```